specs/chain/hs/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -17,18 +18,39 @@ import Ledger.Core
 import Ledger.Delegation
 import Ledger.Signatures
 
-data BlockHeader
-  = BlockHeader
-  { _prevHHash :: !Hash
-    -- ^ Hash of the previous block header, or 'genesisHash' in case of the
-    -- first block in a chain.
-  , _bSlot :: !Slot
-    -- ^ Absolute slot for which the block was generated.
-  , _bIssuer :: !VKey
-    -- ^ Block issuer.
-  , _bSig :: !(Sig VKey)
-    -- ^ Signature of the block by its issuer.
+-- TODO(md): At the moment there is no description of the type in the
+-- LaTeX specification.
+data BHToSign deriving (Eq, Show, Generic)
+
+data BlockProtocolHeader
+  = MkBlockProtocolHeader
+  {
+    -- | Hash of the previous block header, or 'genesisHash' in case of
+    -- the first block in a chain.
+    _bhPrevHash :: Hash
+    -- | Hash of the block.
+  , _bhHash     :: Hash
+    -- | Block issuer.
+  , _bhIssuer   :: VKey
+    -- | Part of the block header which must be signed.
+  , _bhToSign   :: BHToSign
+    -- | Signature of the block by its issuer.
+  , _bhSig      :: (Sig VKey)
+  } deriving (Eq, Show, Generic)
+
+makeLenses ''BlockProtocolHeader
 
+
+-- Actually, there's nothing that is exclusive to the payload part of
+-- the header so no need for this data structure
+data BlockPayloadHeader
+
+data BlockHeader
+  = MkBlockHeader
+  { -- | Absolute slot for which the block was generated.
+    _bhSlot :: Slot
+    -- | Part of the header specific to the Permissive BFT protocol.
+  , _bhPBFT :: BlockProtocolHeader
```

Comment:
No need for this intermediate structure, just fold everything in to `BlockHeader`

Reply:
Sure, I'll do that. It's just that it is not clear to me where in data structures to reflect the split between the payload and protocol specifics. @edsko , does what Nicholas is saying agree with what you had in mind?

Reply:
For our purposes it is useful to separate this out, but it would be possible to do this as a separate step I think if that's more convenient.